mw-webapp/src/hooks/useWayPreview.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from "react";
+import {useNavigate} from "react-router-dom";
+import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
+import {WayPreview} from "src/model/businessModelPreview/WayPreview";
+import {pages} from "src/router/pages";
+
+/**
+ * Fetches WayPreview and returns state
+ */
+export function useWayPreview (uuid: string) {
+  const [way, setWay] = useState<WayPreview | null>(null);
+  const navigate = useNavigate();
+
+  /**
+   * Get WayPreview
+   */
+  const loadWay = async () => {
+    const data = await WayPreviewDAL.getWayPreview(uuid);
```

Comment:
data is too abstract name for variable, could we specify if here?

Reply:
Renamed to wayData